src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -823,6 +823,56 @@ def deviceAvailable(self, device, err):
 
         return DeviceProxyWrapper(returnDevice, self._dmLib)
 
+    async def GetConnectedDevice(self, nodeid, allowPASE=True, timeoutMs: int = None):
+        ''' Returns DeviceProxyWrapper upon success.'''
+        self.CheckIsActive()
+
+        if allowPASE:
+            returnDevice = c_void_p(None)
+            res = self._ChipStack.Call(lambda: self._dmLib.pychip_GetDeviceBeingCommissioned(
+                self.devCtrl, nodeid, byref(returnDevice)), timeoutMs)
+            if res.is_success:
+                logging.info('Using PASE connection')
+                return DeviceProxyWrapper(returnDevice)
+
+        eventLoop = asyncio.get_running_loop()
+        future = eventLoop.create_future()
+
+        class DeviceAvailableClosure():
+            def __init__(self, loop, future: asyncio.Future):
+                self._returnDevice = c_void_p(None)
+                self._returnErr = None
+                self._event_loop = loop
+                self._future = future
+
+            def _deviceAvailable(self):
+                if self._returnDevice.value is not None:
+                    self._future.set_result(self._returnDevice)
+                else:
+                    self._future.set_exception(self._returnErr.to_exception())
```

Comment:
for easier debugging, should we handle the case of `obj.deviceAvailable(None, None)` ever being called? In that case _returnErr would be none and we would not have to_exception. Should we add at least a log below in deviceAvailable or make the exception here something like generic `No device but no exception either` ?

Reply:
As it stands now, the second argument will always be a valid object (generated by `ToPyChipError`). If we ever change the semantics, it would throw a `AttributeError` here indeed, but I don't think that is bad. In fact, it would help debugging quite nicely as it prints a nice stack trace :sweat_smile: I'd prefer to apply offensive programming here, and leave it as is.